language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/filters/PackageActionFunctionAndTypesFilter.java

```diff
@@ -259,21 +260,16 @@ private boolean isConnectorInit(int startIndex, TextDocumentServiceContext conte
      * @return {@link Integer}      Index of the delimiter
      */
     private int getPackageDelimiterTokenIndex(TextDocumentServiceContext completionContext) {
-        ArrayList<String> terminalTokens = new ArrayList<>(Arrays.asList(new String[]{";", "}", "{"}));
+        ArrayList<String> terminalTokens = new ArrayList<>(Arrays.asList(new String[]{";", "}", "{", "(", ")"}));
+        int delimiterIndex = -1;
         int searchTokenIndex = completionContext.get(DocumentServiceKeys.TOKEN_INDEX_KEY);
         TokenStream tokenStream = completionContext.get(DocumentServiceKeys.TOKEN_STREAM_KEY);
-        int delimiterIndex = -1;
-        String currentTokenStr = tokenStream.get(searchTokenIndex).getText();
-
-        if (terminalTokens.contains(currentTokenStr)) {
-            searchTokenIndex -= 1;
-            while (true) {
-                if (tokenStream.get(searchTokenIndex).getChannel() == Token.DEFAULT_CHANNEL) {
-                    break;
-                } else {
-                    searchTokenIndex -= 1;
-                }
-            }
+        /*
+        In order to avoid the token index inconsistencies, current token index offsets from two default tokens
+         */
+        Token offsetToken = CommonUtil.getNthDefaultTokensToLeft(tokenStream, searchTokenIndex, 2);
+        if (!terminalTokens.contains(offsetToken.getText())) {
+            searchTokenIndex = offsetToken.getTokenIndex();
```

Comment:
this is same code as the file above. Is there a way to reuse these as a utils?

Reply:
This section is still improving. So the logic can be different with the new use-cases. So ATM I think it would be better we keep this as it is, otherwise, in the future, we might need to revert it back